docs/concepts.md

```diff
@@ -67,6 +68,118 @@ that are specific to a particular Node (like per-process UNIX file descriptors),
 so handle value 3 for Node A identify a different channel than handle value 3
 for Node B (unless there happens to be a coincidence of numbering).
 
+## Labels
+
+Labels are used to enforce information flow control (IFC) between nodes and
+channels in an Oak Application and the outside world.
+
+Each node and each channel is associated with a label at creation time. The
+label does not change any more after that (apart from explicit declassification
+operations, which are not currently supported).
+
+A label has two components: **secrecy** and **integrity**.
+
+Each component is represented as an (unordered) set of tags, and each tag refers
+to individual security principal in the system.
+
+The following types of security principals are supported in Oak:
+
+- **user**: specified by a bearer token, public key, or some other assertion of
+  a user identity
+- **computation**: specified by a measurement (hash) of the WebAssembly module
+  running within an Oak Node
```

Comment:
What's the mapping between secrecy and integrity components, and security principals?  Any to any, or are (say) "user" security principals always associated with secrecy, not integrity?

Reply:
The system allows all combinations, but it's up to developers / users to determine whether they care about e.g. whether a piece of data actually comes from a user or is the result of a computation, in which case a user or computation tag may be used for integrity.